tests/std/tests/P0896R4_ranges_alg_transform_binary/test.cpp

```diff
@@ -0,0 +1,154 @@
+            int output[3] = {-1, -1, -1};
+            Read1 wrapped_in1{input1};
+            Read2 wrapped_in2{input2};
+
+            auto result = transform(wrapped_in1.begin(), wrapped_in1.end(), wrapped_in2.begin(), wrapped_in2.end(),
+                Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out.peek() == output + 3);
+            assert(ranges::equal(output, expected));
+        }
+        { // Validate range overload
+            int output[3] = {-1, -1, -1};
+            Read1 wrapped_in1{input1};
+            Read2 wrapped_in2{input2};
+
+            auto result = transform(wrapped_in1, wrapped_in2, Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out.peek() == output + 3);
+            assert(ranges::equal(output, expected));
+        }
+        { // Validate range overload first range shorter
+            int output[2] = {-1, -1};
+            Read1 wrapped_in1{shortInput1};
+            Read2 wrapped_in2{input2};
+
+            auto result = transform(wrapped_in1, wrapped_in2, Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(result.in1 == wrapped_in1.end());
+            assert(next(result.in2) == wrapped_in2.end());
+            assert(result.out.peek() == output + 2);
+            assert(ranges::equal(output, shortExpected));
+        }
+        { // Validate range overload second range shorter
+            int output[2] = {-1, -1};
+            Read1 wrapped_in1{input1};
+            Read2 wrapped_in2{shortInput2};
+
+            auto result = transform(wrapped_in1, wrapped_in2, Write{output}, plus{}, get_first, get_second);
+            STATIC_ASSERT(
+                same_as<decltype(result), binary_transform_result<iterator_t<Read1>, iterator_t<Read2>, Write>>);
+            assert(next(result.in1) == wrapped_in1.end());
+            assert(result.in2 == wrapped_in2.end());
+            assert(result.out.peek() == output + 2);
+            assert(ranges::equal(output, shortExpected));
+        }
+    }
+};
+
+using Elem1 = const P;
+using Elem2 = const P;
+using Elem3 = int;
+
+#ifdef TEST_EVERYTHING
+int main() {
+    // No constexpr test here; the test_in_in_write call exceeds the maximum number of steps in a constexpr computation.
+    test_in_in_write<instantiator, Elem1, Elem2, Elem3>();

```

Comment:
Just curious; have you compiled this with `TEST_EVERYTHING` and observed the compiler's memory usage? `in_in_write` is 55 * 55 * 15 = 45375 different permutations. (No change requested)

Reply:
My computer at home runs in the potato class. The poor thing would explode